### PR TITLE
feat(cni): add cni ebpf plugin

### PR DESCRIPTION
**Pull request by bartsmykla**
Closes https://github.com/kumahq/kuma/issues/4584

### Checklist prior to review

- [x] Link to docs PR or issue -- not yet, will follow up with the docs
- [x] Link to UI issue or PR -- no UI changes for now
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS -- CNI works only on windows
- [x] Unit Tests -- no unit tests as the changes are only in helm chart templates
- [x] E2E Tests -- no e2e tests yet, will follow up with the e2e tests in another PR
- [x] Manual Universal Tests -- CNI works only on Kubernetes
- [x] Manual Kubernetes Tests -- locally on KinD and on GKE cluster
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? -- no
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? -- no

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by lukidzi — APPROVED** [review]

File: deployments/charts/kuma/templates/cni-daemonset.yaml [hunk 1]
```diff
@@ -51,8 +51,27 @@ spec:
       containers:
         - name: install-cni
           {{- if .Values.experimental.cni }}
+          {{- if .Values.experimental.ebpf.enabled }}
+          image: {{ include "kuma.formatImage" (dict "image" .Values.cni.experimental.imageEbpf "root" $) | quote }}
```
Comment (jakubdyszkiewicz) [inline deployments/charts/kuma/templates/cni-daemonset.yaml:55]: I don't think we should use `kuma.formatImage` in this place.
It does some defaults from `global` etc. which we don't want to do. I think we should just explicitly do
```
{{- printf "%s/%s:%s" $registry $repo $tag -}}
```

File: deployments/charts/kuma/templates/cni-rbac.yaml [hunk 2]
```diff
@@ -21,10 +21,16 @@ metadata:
...
       - nodes
     verbs:
       - get
+  - apiGroups: [""]
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (jakubdyszkiewicz) [inline deployments/charts/kuma/templates/cni-rbac.yaml:33]: I think I'd add list and watch for nodes too for simplicity, but up to you

Comment (bartsmykla) [reply deployments/charts/kuma/templates/cni-rbac.yaml:33]: I'm a fan of least privileges approach, and as merbridge currently doesn't need these permissions I would like to leave it as is

Comment (jakubdyszkiewicz) [reply deployments/charts/kuma/templates/cni-rbac.yaml:33]: fair enough

File: deployments/charts/kuma/templates/cni-daemonset.yaml (hunk 1 above)
Comment (bartsmykla) [reply deployments/charts/kuma/templates/cni-daemonset.yaml:55]: ack

Comment (bartsmykla) [reply deployments/charts/kuma/templates/cni-daemonset.yaml]: addressed

File: app/kumactl/cmd/install/testdata/install-control-plane.cni-enabled.golden.yaml [hunk 3]
```diff
@@ -1977,7 +1981,7 @@ spec:
       containers:
         - name: install-cni
           image: "docker.io/kumahq/install-cni:0.0.10"
-          imagePullPolicy: Always
+          imagePullPolicy: IfNotPresent
```
Comment (slonka) [inline app/kumactl/cmd/install/testdata/install-control-plane.cni-enabled.golden.yaml:1984]: why are we changing the old image policy? I think that `IfNotPresent` is correct but why was it `Always` before?

File: deployments/charts/kuma/templates/cni-daemonset.yaml [hunk 4]
```diff
@@ -61,16 +82,23 @@ spec:
...
           {{- else }}
           image: {{ include "kuma.formatImage" (dict "image" .Values.cni.image "root" $) | quote }}
-          imagePullPolicy: Always
+          imagePullPolicy: {{ .Values.cni.image.imagePullPolicy }}
           command: [ "/bin/sh", "-c", "--" ]
           args: [ "sleep {{.Values.cni.delayStartupSeconds}} && exec /install-cni.sh" ]
           {{- end }}
+          {{- if and .Values.experimental.cni .Values.experimental.ebpf.enabled }}
+          securityContext:
+            privileged: true
```
Comment (slonka) [inline deployments/charts/kuma/templates/cni-daemonset.yaml:94]: it needs to be `privileged` right?

File: deployments/charts/kuma/templates/cni-rbac.yaml [hunk 5]
```diff
@@ -21,10 +21,18 @@ metadata:
 rules:
   - apiGroups: [""]
     resources:
-      - pods
       - nodes
     verbs:
       - get
+  - apiGroups: [""]
+    resources:
```
Comment (slonka) [inline deployments/charts/kuma/templates/cni-rbac.yaml:28]: why do we need more permissions?

**Review by slonka — APPROVED** [review]
Overall looks ok, a couple of questions just

File: app/kumactl/cmd/install/testdata/install-control-plane.cni-enabled.golden.yaml (hunk 3 above)
Comment (bartsmykla) [reply app/kumactl/cmd/install/testdata/install-control-plane.cni-enabled.golden.yaml:1984]: I don't know, but it was just wrong, and now can be configured if you want (wasn't possible before)

File: deployments/charts/kuma/templates/cni-daemonset.yaml (hunk 4 above)
Comment (bartsmykla) [reply deployments/charts/kuma/templates/cni-daemonset.yaml:94]: true

File: deployments/charts/kuma/templates/cni-rbac.yaml (hunk 5 above)
Comment (bartsmykla) [reply deployments/charts/kuma/templates/cni-rbac.yaml:28]: merbridge needs it for its localip controller (ie. https://github.com/merbridge/merbridge/blob/36ee27af5021bda9e3894278af5b40542e9a3dc8/controller/localip/localip.go#L40)

**Pull request merged**